File: JavaScript/Factorial_Recursive.js
//Program to find the factorial of a number recursively

//prompts the user and inputs the number 
num1 = prompt("This program finds the factorial of a number. \n Which number would you like to input? ")


function fact(num1) {
    if (num1 === 1) {
        return 1;
    }
    return num1 * fact(num1 - 1);
}


factorial_total = fact(num1)

print("The factorial of ")
print(n)
print(" is ")
print(factorial_total)

File: C++/Factorial_recursive.cpp
#include <iostream>

int main() {

    int n = 0;
    std::cout << "This program finds the factorial of a number.\n" << std::endl;
    std::cout << "Which number would you like to find?\n" << std::endl;
    std::cin >> n;

    std::cout << "The factorial of " << n << " is " << factorial(n) << std::endl;

    return 0;
}


int factorial(int n) {
    if(n == 1)
        return 1;
    else
        return n * factorial(n - 1);
}